### ISSUE TITLE
RFE: devolve rich dependencies to first dependency encounter with --norichdeps

### ISSUE THREAD

**Issue by n3npq** [rpm-software-management/rpm#461]
Related to preventing rich depsolver from being used.

To illustrate the RFE, the rpm-ecosystem mailing list reports a failure parsing

Requires: (annobin if gcc)

Devolving that dependency to another known form, perhaps a simple Suggests/Requires,  seems a possible approach to compatibility.

Simple alternation of (A || B) could -- under an assumption that the likeliest predicate is used first -- could be transformed to A.

Devolving logical expressions to simpler predicates used by rpm can/will never be perfect.

However devolving to simpler predicates might be sufficient to bootstrap depsolvers and other tools with enough compatibility to be able to function in spite of the tools being incapable of parsing rich dependencies.

**Comment by ignatenkobrain** [comment]
This is pretty much useless.

Requires: (annobin if gcc) is not same as `Recommends: annobin`. Requires: annobin sounds a bit better, but not enough.

Solving forward-compatibility is out of scope. CLOSING.
